src/Components/CardboardModal/CardboardModal.tsx

```diff
@@ -0,0 +1,129 @@
+
+const stackTokens: IStackTokens = { childrenGap: 8 };
+
+const getClassNames = classNamesFunction<
+    ICardboardModalStyleProps,
+    ICardboardModalStyles
+>();
+
+const CardboardModal: React.FC<ICardboardModalProps> = (props) => {
+    const {
+        isOpen,
+        onDismiss,
+        primaryButtonProps,
+        title,
+        children,
+        destructiveButtonProps,
+        styles,
+        subTitle,
+        titleIconName
+    } = props;
+
+    // contexts
+
+    // state
+
+    // hooks
+    const { t } = useTranslation();
+    const titleId = useId('modal-title');
+
+    // callbacks
+
+    // side effects
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme(),
+        isDestructiveFooterActionVisible: !!destructiveButtonProps
+    });
+
+    return (
+        <Modal
+            isOpen={isOpen}
+            titleAriaId={titleId}
+            onDismiss={onDismiss}
+            styles={classNames.subComponentStyles.modal}
+        >
+            <Stack tokens={stackTokens} style={{ height: '100%' }}>
+                <div className={classNames.headerContainer}>
+                    <div className={classNames.titleContainer}>
+                        {titleIconName && (
+                            <Icon
+                                iconName={titleIconName}
+                                styles={classNames.subComponentStyles.icon}
+                            />
+                        )}
+                        <h3 id={titleId} className={classNames.title}>
+                            {typeof title === 'function' ? title() : title}
+                        </h3>
+                    </div>
+                    {subTitle && (
+                        <p className={classNames.subtitle}>
+                            {typeof subTitle === 'function'
+                                ? subTitle()
+                                : subTitle}
+                        </p>
+                    )}
+                </div>
+                <div className={classNames.content}>
+                    <Stack tokens={stackTokens}>{children}</Stack>
+                </div>
+                <div className={classNames.footer}>
+                    <Stack
+                        horizontal
+                        tokens={stackTokens}
+                        styles={classNames.subComponentStyles.footerStack}
+                    >
+                        {destructiveButtonProps && (
+                            <PrimaryButton
+                                {...destructiveButtonProps}
+                                styles={classNames.subComponentStyles.destructiveButton?.()}
+                            />
+                        )}
+                        <Stack
+                            tokens={stackTokens}
+                            horizontal
+                            horizontalAlign={'end'}
+                        >
+                            <PrimaryButton
```

Comment:
I think the preference we have is placing the primary button to the right and secondary/default button to the left

Reply:
Updated.